pkg/client/context.go

```diff
@@ -0,0 +1,61 @@
+package client
+
+import (
+	"context"
+	"net/http"
+)
+
+type contextKey int
+
+const (
+	authKey contextKey = iota
+	williamKey
+)
+
+func Auth(r *http.Request) *AuthInfo {
+	if r == nil {
+		return nil
+	}
+
+	return auth(r)
+}
+
+func auth(r *http.Request) *AuthInfo {
+	if a := r.Context().Value(authKey); a != nil {
+		return a.(*AuthInfo)
+	}
+
+	return nil
+}
+
+func Get(r *http.Request) William {
+	if r == nil {
+		return nil
+	}
+
+	return get(r)
+}
+
+func get(r *http.Request) William {
```

Comment:
What led you to use this pattern of public/private methods? Any advantages over placing the whole logic inside `Get`, for instance?

Reply:
The idea of go is to leave less code in public methods.

Reply:
I agree with Gerson, just stick the body on Get, this method isn't pulling its weight.